### PR TITLE
Make `error.grouping_name` script compatible with synthetic _source

### PR DESCRIPTION
**Pull request by felixbarny**
Makes the indexed runtime field script for `error.grouping_name` compatible with synthetic _source.

### CODE REVIEW

File: x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_error_grouping.yml [hunk 1]
```diff
@@ -39,18 +39,23 @@ setup:
           - create: {}
           - '{"@timestamp": "2017-06-22", "error": {"log": {"message": ""}, "exception": [{"message": "exception_used"}]}}'
 
+          # Non-empty error.exception.message used from array
+          - create: {}
+          - '{"@timestamp": "2017-06-22", "error": {"log": {"message": ""}, "exception": [{"message": "first_exception_used"}, {"message": "2_ignored"}]}}'
```
Comment (lahsivjar) [inline x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_error_grouping.yml:44]:
```suggestion
          - '{"@timestamp": "2017-06-22", "error": {"log": {"message": ""}, "exception": [{"message": ["first_exception_used", "2_ignored"]}]}}'
```
Not sure if I am reading it wrong but it seems that we are adding the code for a message that could be a slice.

Comment (felixbarny) [reply x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_error_grouping.yml:44]: When using synthetic source, the source gets synthesized on-the-fly when accessing `params['_source']` in the runtime field script. The synthesized `_source` looks a bit differently compared to the original one: `{"error":{"exception":"message":["first_exception_used"," 2_ignored"]}}`.

Comment (lahsivjar) [reply x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_error_grouping.yml:44]: Thanks for the explanation!

**Review by lahsivjar — APPROVED** [review]
I think the name of the PR (and the commit) should update `error.grouping_key` -> `error.grouping_name`, rest LGTM!

**Review by carsonip — APPROVED** [review]

**Pull request merged**